Brokerages/Zerodha/ZerodhaLiveOptionChainProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Brokerages.Zerodha
+{
+    public class ZerodhaLiveOptionChainProvider : IOptionChainProvider
+    {
+        private readonly ZerodhaSymbolMapper symbolMapper;
+
+        public ZerodhaLiveOptionChainProvider(ZerodhaSymbolMapper _symbolMapper)
+        {
+            symbolMapper = _symbolMapper;
+        }
+
+        public IEnumerable<Symbol> GetOptionContractList(Symbol symbol, DateTime date)
+        {
+            return symbolMapper.KnownSymbols.Where(s => s.SecurityType == SecurityType.Option && s.ID.Symbol == symbol.Underlying.Value);
```

Comment:
The symbolMapper.KnownSymbols list seems to be populated only once, in the symbol mapper's constructor.
What happens when option contracts expire and new ones are listed?

Might need to get a fresh list of contracts with the Kite API ?